stable/test/test_dep.pony

```diff
@@ -0,0 +1,107 @@
+use "ponytest"
+use "collections"
+use "files"
+use "json"
+use ".."
+
+class TestDep is UnitTest
+  new iso create() => None
+  fun name(): String => "stable.Dep"
+
+  fun apply(h: TestHelper) ? =>
+    let path = FilePath(h.env.root as AmbientAuth, "stable/test/testdata/empty-deps")?
+    let bundle = Bundle(path, LogNone, false)?
+    var dep: DepAny
+
+    var info: JsonObject box = JsonObject()
+    h.assert_error({ ()? => Dep(bundle, info)? }, "empty JSON")
+
+    // minimal github dep
+    let github = Map[String, JsonType]
+    github("type") = "github"
+    github("repo") = "foo/bar"
+
+    info = JsonObject.from_map(github)
+    dep = Dep(bundle, info)?
+    h.assert_eq[String]("https://github.com/foo/bar", (dep as DepGitHosted).url())
+    h.assert_eq[String](path.join(".deps/foo/bar")?.path, dep.root_path())
+    h.assert_eq[String](path.join(".deps/foo/bar")?.path, dep.packages_path())
+
+    // complete github dep
+    github("subdir") = "baz"
+    github("tag") = "v1"
```

Comment:
I was surprised about that, too, but it's no change that's perceivable using anything getter-like, IIRC. 

For example, with [`DepGitHosted`](https://github.com/ponylang/pony-stable/blob/07ce8c8399157f6e049e670d81caeb7f7f1f2b81/stable/dep.pony#L18-L64), the only moment when that tag becomes effective is when checking out the sources.

I suppose one way to test this could be to introduce some mock for _not actually shelling out_; or adding integration tests for `stable fetch` (#28).